pkg/apis/extensions/types.go

```diff
@@ -82,29 +82,48 @@ type SubresourceReference struct {
 	Subresource string `json:"subresource,omitempty"`
 }
 
-// ResourceConsumption is an object for specifying average resource consumption of a particular resource.
-type ResourceConsumption struct {
-	Resource api.ResourceName  `json:"resource,omitempty"`
-	Quantity resource.Quantity `json:"quantity,omitempty"`
+// MetricName is an identifier of a Heapster metric.
+type MetricName string
+
+const (
+	// Heapster metric representing CPU usage of a pod (in milli-cores).
+	MetricCPUUsage MetricName = "kubernetes.io:cpu-usage"
+	// Heapster metric representing Memory usage of a pod (in bytes).
+	MetricMemoryUsage MetricName = "kubernetes.io:memory-usage"
+)
+
+// MetricUtilization is an object for specifying the arthmetic mean utilization of a particular resource.
+type MetricUtilization struct {
+	// Metric specifies either the identifier of the target metric when present in the spec, or the identifier of the observed metric when present in the status.
+	Metric MetricName `json:"metric"`
+	// Utilization specifies either the target arthmetic mean utilization of the metric when present in the spec, or the observed arthmetic mean utilization when present in the status.
```

Comment:
In my opinion having a single struct brings benefits: we are not duplicating code. I don't think this causes much confusion.